packages/ldc.rb

```diff
@@ -0,0 +1,39 @@
+require 'package'
+
+class Ldc < Package                 # The first character of the class name must be upper case
+  description 'D language compiler using LLVM.'
+  homepage 'https://github.com/ldc-developers/ldc'
+  version '1.7.0'
+  source_url 'https://github.com/ldc-developers/ldc/releases/download/v1.7.0/ldc-1.7.0-src.tar.gz'
+  source_sha256 '7cd46140ca3e4ca0d52c352e5b694d4d5336898ed4f02c3e18e0eafd69dd18bd'
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  depends_on 'llvm'
+  depends_on 'curl'
+  depends_on 'gcc'
+  depends_on 'ncurses'
+  depends_on 'zlibpkg'
+  depends_on 'libconfig' => :build
+  depends_on 'cmake' => :build
+  depends_on 'libedit' => :build
+
+  def self.build                   # the steps required to build the package
+    system "mkdir", "build"
+    system "curl -L https://github.com/ldc-developers/ldc/releases/download/v0.17.5/ldc-0.17.5-src.tar.gz \
+            | tar xzf - -C build"
```

Comment:
I guess I'm not understanding why this step is needed.  Isn't this the same project but a different version?

Reply:
Because this version is last C++ version, so it is needed for bootstraping, because current ldc codebase is written in D language itself, so there is a need of some D compiler to compile it.

Reply:
Thanks for the explanation.  That makes sense.  We should probably do a checksum on this tarball just to make sure it downloaded correctly.  Something like:
```
abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('ldc-0.17.5-src.tar.gz') ) == '12345abcde...'
```

Reply:
Thats a good idea. I will add it